File: Devlab/932.cpp
/* https://www.borntodev.com/devlab/task/932 */

#include <bits/stdc++.h>
using namespace std;

int main()
{
	  unsigned long long a,b,MOD = 1e9+7;
    cin>>a>>b;
    cout<<((((unsigned long long)(floor(log10(a)*b)))+1)%MOD);
}

File: Devlab/932.py
## https://www.borntodev.com/devlab/task/932 ##
## Python sol

import math

a,b = [int(input()) for i in range(2)]
MOD = 1000000007
ans = ( ( math.floor( (math.log10(a)*b) ) + 1 ) % MOD)
print(int(ans))
